pkg/querier/distributor_queryable.go

```diff
@@ -84,8 +84,12 @@ func (q *distributorQuerier) streamingSelect(sp storage.SelectParams, matchers [
 		return nil, nil, promql.ErrStorage{Err: err}
 	}
 
-	serieses := make([]storage.Series, 0, len(results))
-	for _, result := range results {
+	if len(results.Timeseries) != 0 {
```

Comment:
I had recently change the QueryStreamResponse to use a `oneof` label so that a response could only return TimeSeries or ChunkSeries. However that came back to the distributor, which combines different return types, which comes back to this, where even though we may have two return types, we only return one type. 

Allowing the distributor to return an conjoined set of ChunkSeries and TimeSeries is more complicated, but potentially allows a easier migration path from old chunk format to block format. Since a querier could be deployed that supports both ingesters. But it will require more complexity here since I believe we'll need to do some merging between the two sets. 

Is that potential migration path the one we might expect to go? Or am I over thinking this, and we'd require a hard cutover of ingesters with streaming, in which case I can reimplement the oneof behavior and returning whichever payload here would be acceptable. @pstibrany @pracucci

Reply:
> Is that potential migration path the one we might expect to go?

No. The support for `Timeseries` is temporary. We plan to drop it at some point, as soon as TSDB will allow to directly query chunks (there's an on-going work from Bartek). The reason is that we don't want to pay the CPU cost to decode chunks in the ingesters but only in the queriers, so we want to ship TSDB chunks from the ingesters to the querier like we do for the chunks storage and then decode the chunks in the querier. It's not possible to do yet, but that's the final goal.

Reply:
Sounds good. I'll reimplement with oneof then, and have it conditionally only return a single type